packages/frontend/src/components/navigation/ConnectLedger.js

```diff
@@ -24,11 +29,17 @@ const ConnectLedgerButton = styled.div`
     }
 `;
 
-export default ({
-    connectLedger,
-    ledgerConnectionAvailable,
-    showConnectLedgerButton
-}) => {
+export default () => {
+    const dispatch = useDispatch();
+
+    const { pathname } = useSelector(getLocation);
+    const ledgerConnectionAvailable = useSelector(selectLedgerConnectionAvailable);
+    const hasLedger = useSelector(selectLedgerHasLedger);
+
+    const connectLedger =  () => dispatch(handleShowConnectModal());
+    
+    const showConnectLedgerButton = hasLedger || ['sign-in-ledger', 'setup-ledger'].includes(pathname.split('/')[1]);
+
     return showConnectLedgerButton
```

Comment:
💄 Could we short-circuit the return here instead of returning a ternary?

i.e.
```js
if (!showConnectionLedgerButton) {
  return null;
}

return (
  ...
);
```

Reply:
Done.